dotnet/src/Azure.Iot.Operations.Connector/ConnectorWorker.cs

```diff
@@ -393,44 +390,39 @@ public async Task ForwardReceivedEventAsync(string deviceName, string inboundEnd
             {
                 if (destination.Target == EventStreamTarget.Mqtt)
                 {
-                    string topic = destination.Configuration.Topic ?? throw new AssetConfigurationException($"Dataset with name {assetEvent.Name} in asset with name {asset.DisplayName} has no configured MQTT topic to publish to. Data won't be forwarded for this dataset.");
-                    var mqttMessage = new MqttApplicationMessage(topic)
+                    string topic = destination.Configuration.Topic ??
+                                   throw new AssetConfigurationException(
+                                       $"Dataset with name {assetEvent.Name} in asset with name {asset.DisplayName} has no configured MQTT topic to publish to. Data won't be forwarded for this dataset.");
+
+                    var messageMetadata = new OutgoingTelemetryMetadata
                     {
-                        PayloadSegment = serializedPayload,
+                        CloudEvent = cloudEvent
                     };
 
                     Retain? retain = destination.Configuration.Retain;
                     if (retain != null)
                     {
-                        mqttMessage.Retain = retain == Retain.Keep;
-                    }
-
-                    if (cloudEvent != null)
-                    {
-                        mqttMessage.AddCloudEvents(cloudEvent);
+                        messageMetadata.Retain = retain == Retain.Keep;
                     }
 
                     if (userData != null)
                     {
                         foreach (string key in userData.Keys)
                         {
-                            mqttMessage.AddUserProperty(key, userData[key]);
+                            messageMetadata.UserData[key] = userData[key];
                         }
                     }
 
-                    MqttClientPublishResult puback = await _mqttClient.PublishAsync(mqttMessage, cancellationToken);
+                    var telemetrySender = _telemetrySenderCache.GetOrAdd(topic, t => new ConnectorTelemetrySender(_applicationContext, _mqttClient, t));
+                    await telemetrySender.SendTelemetryAsync(
+                        serializedPayload,
+                        messageMetadata,
+                        null,
+                        MqttQualityOfServiceLevel.AtLeastOnce,
```

Comment:
The QOS may be 0 or 1 here. Use the value from ```destination.Configuration.QoS``` to determine the value for this argument